### PR TITLE
ci(github-actions): add action to force backport labels

### PR DESCRIPTION
**Pull request by fruch**
we have it on scylla core repo, and we should have it here in SCT as well, so people open PRs would need to specify backport before it can be merged

### Testing

- [ ]

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: .github/workflows/pr-require-backport-label.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: PR require backport label
+on:
+  pull_request:
+    types: [opened, labeled, unlabeled, synchronize]
+    branches:
+      - master
+      - next
```
Comment (vponomaryov) [inline .github/workflows/pr-require-backport-label.yaml:7]: We don't have "next"

File: .github/workflows/pr-require-backport-label.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: mheap/github-action-required-labels@v5
+        with:
+          mode: minimum
+          count: 1
+          labels: "backport/none\nbackport/\\d.\\d"
```
Comment (vponomaryov) [inline .github/workflows/pr-require-backport-label.yaml:20]: This regex won't work on some of our labels such as `backport/manager` and `backport/perf-v14`.

File: .github/workflows/pr-require-backport-label.yaml (hunk 1 above)
Comment (fruch) [reply .github/workflows/pr-require-backport-label.yaml:7]: removed

File: .github/workflows/pr-require-backport-label.yaml (hunk 2 above)
Comment (fruch) [reply .github/workflows/pr-require-backport-label.yaml]: replaced it with a more fitting one.

File: .github/workflows/pr-require-backport-label.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: mheap/github-action-required-labels@v5
+        with:
+          mode: minimum
+          count: 1
+          labels: "backport/none\nbackport/.*-v?[\\d.]*$"
```
Comment (vponomaryov) [inline .github/workflows/pr-require-backport-label.yaml:20]: It won't catch the `backport/2024.1` label

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-require-backport-label.yaml (hunk 3 above)
Comment (fruch) [reply .github/workflows/pr-require-backport-label.yaml]: I think now it's a better version of the regex, that does cover all of the options

File: .github/workflows/pr-require-backport-label.yaml (hunk 1 above)
Comment (vponomaryov) [reply .github/workflows/pr-require-backport-label.yaml:7]: Not removed.

Comment (fruch) [reply .github/workflows/pr-require-backport-label.yaml]: now it's removed

**Review by vponomaryov — APPROVED** [review]
LGTM

**Pull request merged**